cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/ReducerTransformExecutorFactory.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.etl.batch.mapreduce;
+
+import co.cask.cdap.api.dataset.lib.KeyValue;
+import co.cask.cdap.api.mapreduce.MapReduceTaskContext;
+import co.cask.cdap.api.metrics.Metrics;
+import co.cask.cdap.etl.api.Aggregator;
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.cdap.etl.api.Transformation;
+import co.cask.cdap.etl.api.batch.BatchAggregator;
+import co.cask.cdap.etl.api.batch.BatchRuntimeContext;
+import co.cask.cdap.etl.batch.PipelinePluginInstantiator;
+import co.cask.cdap.etl.batch.TransformExecutorFactory;
+import co.cask.cdap.etl.batch.conversion.IdentityConversion;
+import co.cask.cdap.etl.batch.conversion.WritableConversion;
+import co.cask.cdap.etl.batch.conversion.WritableConversions;
+import co.cask.cdap.etl.common.DatasetContextLookupProvider;
+import com.google.common.base.Function;
+import com.google.common.collect.Iterators;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.io.WritableComparable;
+import org.apache.hadoop.mapreduce.Reducer;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Creates transform executors for mapreduce programs.
+ *
+ * @param <T> the type of input for the created transform executors
+ */
+public class ReducerTransformExecutorFactory<T> extends TransformExecutorFactory<T> {
```

Comment:
this class appears to have a lot of the same code as MapperTransformExecutorFactory. Can they be refactored into one class?